native-sql-engine/cpp/src/codegen/arrow_compute/ext/expression_codegen_visitor.cc

```diff
@@ -246,6 +246,27 @@ arrow::Status ExpressionCodegenVisitor::Visit(const gandiva::FunctionNode& node)
                << ".rfind(" << child_visitor_list[1]->GetResult()
                << ") != std::string::npos;";
     prepare_str_ += prepare_ss.str();
+  } else if (func_name.compare("like") == 0) {
+    codes_str_ = func_name + "_" + std::to_string(cur_func_id);
+    auto validity = codes_str_ + "_validity";
+    real_codes_str_ = codes_str_;
+    real_validity_str_ = validity;
+    std::stringstream prepare_ss;
+    prepare_ss << GetCTypeString(node.return_type()) << " " << codes_str_ << ";"
+               << std::endl;
+    prepare_ss << "bool " << validity << " = " << child_visitor_list[0]->GetPreCheck()
+               << ";" << std::endl;
+    prepare_ss << "if (" << validity << ") {" << std::endl;
+    prepare_ss << codes_str_ << " = like"
+               << "(" << child_visitor_list[0]->GetResult() << ", "
+               << child_visitor_list[1]->GetResult() << ");" << std::endl;
+    prepare_ss << "}" << std::endl;
+    for (int i = 0; i < 1; i++) {
+      prepare_str_ += child_visitor_list[i]->GetPrepare();
+    }
+    prepare_str_ += prepare_ss.str();
+    check_str_ = validity;
+    header_list_.push_back(R"(#include "precompile/gandiva.h")");
```

Comment:
note: this should be removed after the pre-compile header opt. Will follow up in the next PR